client/src/modules/stock/inventory-adjustment/inventory-adjustment.js

```diff
@@ -0,0 +1,279 @@
+  vm.submit = submit;
+  vm.selectedLots = [];
+  vm.onLotSelect = onLotSelect;
+
+  // grid columns
+  const columns = [
+    {
+      field : 'status',
+      width : 25,
+      displayName : '',
+      cellTemplate : 'modules/stock/exit/templates/status.tmpl.html',
+      enableFiltering : false,
+    },
+
+    {
+      field : 'code',
+      width : 120,
+      displayName : 'TABLE.COLUMNS.CODE',
+      headerCellFilter : 'translate',
+      cellTemplate : 'modules/stock/exit/templates/code.tmpl.html',
+    },
+
+    {
+      field : 'description',
+      displayName : 'TABLE.COLUMNS.DESCRIPTION',
+      headerCellFilter : 'translate',
+      cellTemplate : 'modules/stock/exit/templates/description.tmpl.html',
+      enableSorting : true,
+    },
+
+    {
+      field : 'label',
+      width : 150,
+      displayName : 'TABLE.COLUMNS.LOT',
+      headerCellFilter : 'translate',
+      cellTemplate : 'modules/stock/inventory-adjustment/templates/lot.tmpl.html',
+      enableSorting : true,
+    },
+
+    {
+      field : 'available_lot',
+      width : 150,
+      displayName : 'INVENTORY_ADJUSTMENT.OLD_QUANTITY',
+      headerCellFilter : 'translate',
+      cellTemplate : 'modules/stock/exit/templates/available.tmpl.html',
+      enableFiltering : false,
+    },
+
+    {
+      field : 'quantity',
+      width : 180,
+      displayName : 'INVENTORY_ADJUSTMENT.NEW_QUANTITY',
+      headerCellFilter : 'translate',
+      cellTemplate : 'modules/stock/inventory-adjustment/templates/quantity.tmpl.html',
+      aggregationType : uiGridConstants.aggregationTypes.sum,
+      enableFiltering : false,
+    },
+
+    {
+      field : 'expiration_date',
+      width : 150,
+      displayName : 'TABLE.COLUMNS.EXPIRATION_DATE',
+      headerCellFilter : 'translate',
+      cellTemplate : 'modules/stock/exit/templates/expiration.tmpl.html',
+      enableFiltering : false,
+    },
+  ];
+
+  // grid options
+  vm.gridOptions = {
+    appScopeProvider : vm,
+    enableSorting : true,
+    enableColumnMenus : false,
+    columnDefs : columns,
+    data : vm.Stock.store.data,
+    fastWatch : true,
+    flatEntityAccess : true,
+    rowTemplate : 'modules/templates/grid/error.row.html',
+    onRegisterApi : onRegisterApiFn,
+  };
+
+  // register api
+  function onRegisterApiFn(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  // inline filter
+  vm.toggleInlineFilter = () => {
+    vm.gridOptions.enableFiltering = !vm.gridOptions.enableFiltering;
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.COLUMN);
+  };
+
+  // on lot select
+  function onLotSelect(row) {
+    if (!row.lot || !row.lot.uuid) { return; }
+
+    checkValidity();
+    refreshSelectedLotsList();
+  }
+
+  // update the list of selected lots
+  function refreshSelectedLotsList() {
+    vm.selectedLots = vm.Stock.store.data
+      .filter(item => item.lot && item.lot.uuid)
+      .map(item => item.lot.uuid);
+  }
+
+  // add items
+  function addItems(n) {
```

Comment:
Do we need the `addItems()` function still?

Reply:
nope

Reply:
It is not used in the template but it used to set lines when we load lots